cpp/src/stream_compaction/drop_duplicates.cu

```diff
@@ -0,0 +1,303 @@
+                                     std::vector<null_order>{},
+                                     stream,
+                                     mr);
+
+
+  // extract unique indices 
+  auto device_input_table = cudf::table_device_view::create(keys, stream);
+
+  if(cudf::has_nulls(keys)) {
+    auto comp = row_equality_comparator<true>(*device_input_table,
+                                              *device_input_table,
+                                               nulls_are_equal);
+    auto result_end = unique_copy(rmm::exec_policy(stream)->on(stream),
+        sorted_indices->view().begin<cudf::size_type>(),
+        sorted_indices->view().end<cudf::size_type>(),
+        unique_indices.mutable_view().begin<cudf::size_type>(),
+        comp,
+        keep);
+  
+    return cudf::detail::slice(unique_indices.view(), 0, thrust::distance(unique_indices.mutable_view().begin<cudf::size_type>(), result_end));
+  } else {
+    auto comp = row_equality_comparator<false>(*device_input_table,
+                                               *device_input_table,
+                                               nulls_are_equal);
+    auto result_end = unique_copy(rmm::exec_policy(stream)->on(stream),
+        sorted_indices->view().begin<cudf::size_type>(),
+        sorted_indices->view().end<cudf::size_type>(),
+        unique_indices.mutable_view().begin<cudf::size_type>(),
+        comp,
+        keep);
+  
+    return cudf::detail::slice(unique_indices.view(), 0, thrust::distance(unique_indices.mutable_view().begin<cudf::size_type>(), result_end));
+  }
+  
+}
+
+cudf::size_type unique_count(table_view const& keys,
+                             bool const& nulls_are_equal = true,
+                             rmm::mr::device_memory_resource* mr 
+                                 = rmm::mr::get_default_resource(),
+                             cudaStream_t stream=0)
+{
+  // sort only indices
+  auto sorted_indices = sorted_order(keys,
+                                     std::vector<order>{},
+                                     std::vector<null_order>{},
+                                     stream,
+                                     mr);
+  
+  // count unique elements
+  cudf::size_type* sorted_row_index = const_cast<cudf::size_type *>(sorted_indices->view().data<cudf::size_type>());
+  auto device_input_table = cudf::table_device_view::create(keys, stream);
+
+  if(cudf::has_nulls(keys)) {
+    row_equality_comparator<true> comp (*device_input_table,
+                                              *device_input_table,
+                                              nulls_are_equal);
+    return thrust::count_if(rmm::exec_policy(stream)->on(stream),
+              thrust::counting_iterator<cudf::size_type>(0),
+              thrust::counting_iterator<cudf::size_type>(keys.num_rows()),
+              [sorted_row_index, comp]
+              __device__ (cudf::size_type i) mutable {
```

Comment:
What's with the `mutable`? This lambda doesn't look like it needs to modify anything it is capturing.

Reply:
The equality_comparator operator member function is not a const type, which gives compilation error.

Reply:
Just make it const then.

Reply:
I tried, as it is using begin and end which are non const so it is kind of spiraling down. I will give it a try again.